mm2src/lp_ordermatch.rs

```diff
@@ -2807,43 +2793,67 @@ pub async fn lp_ordermatch_loop(ctx: MmArc) {
                             to_cancel.push(*uuid);
                             continue;
-                        }
+                        },
+                    };
+                    let max_vol =
+                        match calc_max_maker_vol(&ctx, &base, &current_balance, FeeApproxStage::OrderIssue).await {
+                            Ok(max) => max,
+                            Err(e) => {
+                                log::info!("Error {} on balance check to kickstart order {}, cancelling", e, uuid);
+                                to_cancel.push(*uuid);
+                                continue;
+                            },
+                        };
+                    if max_vol < order.available_amount() {
+                        order.max_base_vol = order.reserved_amount() + max_vol;
+                    }
+                    if order.available_amount() < order.min_base_vol {
+                        log::info!("Insufficient volume available for order {}, cancelling", uuid);
+                        to_cancel.push(*uuid);
+                        continue;
+                    }
 
-                        let topic = order.orderbook_topic();
-                        if !ordermatch_ctx.orderbook.lock().await.is_subscribed_to(&topic) {
-                            let request_orderbook = false;
-                            if let Err(e) = subscribe_to_orderbook_topic(
-                                &ctx,
-                                &order.base_orderbook_ticker(),
-                                &order.rel_orderbook_ticker(),
-                                request_orderbook,
-                            )
-                            .await
-                            {
-                                log::error!("Error {} on subscribing to orderbook topic {}", e, topic);
-                            }
-                        }
-                        maker_order_created_p2p_notify(
-                            ctx.clone(),
-                            order,
-                            base.coin_protocol_info(),
-                            rel.coin_protocol_info(),
+                    let topic = order.orderbook_topic();
+                    if !ordermatch_ctx.orderbook.lock().await.is_subscribed_to(&topic) {
+                        let request_orderbook = false;
+                        if let Err(e) = subscribe_to_orderbook_topic(
+                            &ctx,
+                            &order.base_orderbook_ticker(),
+                            &order.rel_orderbook_ticker(),
+                            request_orderbook,
                         )
-                        .await;
+                        .await
+                        {
+                            log::error!("Error {} on subscribing to orderbook topic {}", e, topic);
+                        }
                     }
+                    maker_order_created_p2p_notify(
```

Comment:
Does it mean that we should send the `MakerOrderCreated` message every 0.777 seconds?

Reply:
Same as https://github.com/KomodoPlatform/atomicDEX-API/pull/1179#discussion_r776284882, it's in the `if !ordermatch_ctx.orderbook.lock().await.order_set.contains_key(uuid)` block. It happens only if we have the kickstarted order not broadcast to other nodes yet.